lib/matplotlib/axis.py

```diff
@@ -846,6 +846,13 @@ def get_children(self):
         return [self.label, self.offsetText,
                 *self.get_major_ticks(), *self.get_minor_ticks()]
 
+    # style parameters that should be preserved by reset operations
```

Comment:
This *looks* better, but I'm a bit of a loss what we actually want.

What's the policy for e.g.
- `zorder`
- the grid parameters like `grid_color`

We should have an explicit statement on why we keep some parameters but not others.

Reply:
I intentionally left out grid params because I have no opinion there, but I agree it should be discussed.

Reply:
`zorder` seemed to fit in the "visibility" theme, for which a consensus appears to have been reached. Please correct me if I am wrong.